### PR TITLE
[752] GSuite TXT bug

### PR DESCRIPTION
**Pull request by motdotla**
https://github.com/hackclub/bank/issues/745

### CODE REVIEW

File: config/credentials.yml.enc [hunk 1]
```diff
@@ -1 +1 @@
...
...pYWBy8/3M8yqe9Rm8UrvIEO2qPbadajPtu9FFiKUen0e/+ht0gHfizWX3IpDAQLp7XrI19uSSNEtoe3+bBOHt+FKutx0n5AIYdCWq5N0e1R+Xrki5jyiUQG1UngZ/g0XAA/OaC0GZnPlIHk6llNAbY0musSTQ7cPUW5rZGUC+j9+EBQevSXVpD1jO7GGqvdD0oSEnIj5r12jPEnT/IUrerXs5vprFGgegC2FE8FcDzFFIEd2naNqsatmHiEWl9bsIiFuDGO6N05pIZzXz2td9YePU7IMOep8w/YsNsx79czVTNoH7h+rULosLpGbktNSOL5agZ26qOmApDJ0xr6cVjbpszAdcIrBglHneYh5vVbkO60qq2OHUhrqWmkMCJ5rDlJw4n4lxJJWNMA68ic9a01XTPmmkwjfLvzvJU9/ozRRczm4+x3unXGdvTATH1DrN5Ta3yuhfhwZL8C9Ddy0vcax+XbKVLcIFhqNKELAeWr94DqY6/fGVkK+dCJjKEEead5xSdgMT4qCK53asGYbioICZMYi3kn0C669ZJ62l8jXWnjAkVqmd2+ylJ6QnHcWqWcqh07Ffqh10ljvEibW3uXGv1g4evmv2c2e5jj+OsEOPBAZSJP78qned7SXseo8VR1zZQXUyCuIfI2Yx4uDfjBNVEofdA8vBTpGCiZdp5dDLeTlDfjJwjZBB0yrqtPc2ppXnnMWUbR/KwRw60oarZVZMZJsxrSVvH9jHW+lTWBbu4Go9Jsx5DKXk/mEPT+8zkMFJi3pE308R0nh/m9fnFkpWDNJMLtm88ivXf7hyCE22ekDJkM5iG7yFCYxhI6znGQNOIOZJxXv8T3EWozHJnfoXELiK9Xjw1vA9MIy4VWqfMXOH/kZchUxp2oN/vKBMWOrN5wHB2jeMtfTb5tL+CRC1R4eZ34lwRX2KhfJWwbtN7rEBxT6yd30cFXaYEzf6gO/eBWuiIElkTSKw0TSzoYK97yt18fZTr5GVzoiLmx+kat0z4pREeWFZ/qcMdEE9vPxcLmheJgd1wkyVY0Pm7TVp5y6atw6HhYt0X/Q8D92ogudnolWJjvwuE79zIQYhHIcf8ZnOZ/4Vdd6rzN2mMMV9udS4F3eK6cFgBHKVna7NgLosw0FCxUHHlFXFxla1yQ=--1rlEsZqLP1Yjk2kl--ORxJ0jcp6cdAeT7111CQmA==
```
Comment (motdotla) [inline config/credentials.yml.enc:1]: added gsuite[:customer_id]

File: spec/models/g_suite_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  describe "#verification_url" do
+    it "generates it" do
+      result = g_suite.verification_url
+
+      expect(result).to eql("https://www.google.com/webmasters/verification/verification?siteUrl=http://event1.example.com")
+    end
+  end
+end
+
```
Comment (motdotla) [inline spec/models/g_suite_spec.rb:30]: begins to start adding specs to the project. we can add more as we go.

File: app/services/partners/google/g_suite/domains.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module Partners
+  module Google
+    module GSuite
+      class Domains
+        include Partners::Google::GSuite::Shared::DirectoryClient
+
+        def run
+          directory_client.list_domains(gsuite_customer_id)
```
Comment (motdotla) [inline app/services/partners/google/g_suite/domains.rb:8]: in a future ticket, this method can be used to check which domains are currently verified and which are not. we could produce a nightly report and surface the information to our users - either over email or in their interface or some other way

File: app/controllers/g_suites_controller.rb [hunk 4]
```diff
@@ -36,19 +36,23 @@ def edit
...
       verification_key: g_suite_params[:verification_key],
       dkim_key: g_suite_params[:dkim_key]
-    )
+    }
 
-    authorize @g_suite
-    @g_suite_application = GSuiteApplication.find(g_suite_params[:g_suite_application_id])
-    @g_suite_application.accepted_at = Time.now
-    @g_suite_application.fulfilled_by = current_user
+    @g_suite = GSuiteService::Create.new(attrs).run
```
Comment (motdotla) [inline app/controllers/g_suites_controller.rb:53]: over time, we can start to roll controller logic into their own services - making testing and ultimately management of code easier (especially in these scenarios where a transaction is needed)

**Review by motdotla — COMMENTED** [review]
So after digging, the API I expected to allow us to automate this is for a different purpose (when adding urls/domains to Google Search):

and unfortunately, the DirectoryService API [1] does not support verification either (it does support domain creation):


[1] https://googleapis.dev/ruby/google-api-client/v0.41.1/Google/Apis/AdminDirectoryV1/DirectoryService.html

So ultimately, this PR only adds a link to the verification page on G Suite:

**Review by iamcpdev — APPROVED** [review]

**Pull request merged**